controllers/rhmi/prometheusRules.go

```diff
@@ -19,98 +17,37 @@ var (
-					For:    "10m",
-					Labels: map[string]string{"severity": "warning", "product": installationName, "addon": getAddonName(installation), "namespace": "openshift-monitoring"},
-				},
-				{
-					Alert: fmt.Sprintf("%sUpgradeExpectedDuration60minExceeded", strings.ToUpper(installationName)),
-					Annotations: map[string]string{
-						"sop_url": resources.SopUrlUpgradeExpectedDurationExceeded,
-						"message": fmt.Sprintf("%s operator upgrade is taking more than 60 minutes", strings.ToUpper(installationName)),
-					},
-					Expr:   intstr.FromString(fmt.Sprintf(`%s_version{to_version=~".+",version=~".+"} and (absent((%s_version{job=~"%s.+"} * on(version) csv_succeeded{exported_namespace=~"%s"})) or %s_version)`, installationName, installationName, installationName, installation.Namespace, installationName)),
-					For:    "60m",
-					Labels: map[string]string{"severity": "critical", "product": installationName, "addon": getAddonName(installation), "namespace": "openshift-monitoring"},
-				},
-			},
-		},
-		{
-			AlertName: fmt.Sprintf("%s-missing-metrics", installationName),
+	}
+	err = apiClient.Delete(ctx, installationAlert)
+	if err != nil {
+		return err
+	}
+	upgradeAlert := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-upgrade-alerts", installationName),
 			Namespace: config.OpenshiftMonitoringNamespace,
-			GroupName: fmt.Sprintf("%s-general.rules", installationName),
-			Rules: []monitoringv1.Rule{
-				{
-					Alert: fmt.Sprintf("%sCriticalMetricsMissing", strings.ToUpper(installationName)),
-					Annotations: map[string]string{
-						"sop_url": resources.SopUrlCriticalMetricsMissing,
-						"message": "one or more critical metrics relating to RHOAM installation/upgrade have been missing for 30+ minutes",
-					},
-					Expr:   intstr.FromString(fmt.Sprintf(`absent(%s_version) == 1`, installationName)),
-					For:    "30m",
-					Labels: map[string]string{"severity": "critical"},
-				},
-			},
 		},
-		{
-			AlertName: fmt.Sprintf("%s-telemetry", installationName),
+	}
+	err = apiClient.Delete(ctx, upgradeAlert)
+	if err != nil {
+		return err
+	}
+	metricMissingAlert := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-missing-metrics", installationName),
 			Namespace: config.OpenshiftMonitoringNamespace,
-			GroupName: fmt.Sprintf("%s-telemetry.rules", installationName),
-			Interval:  "30s",
-			Rules: []monitoringv1.Rule{
-				{
-					Expr:   intstr.FromString(fmt.Sprintf("max by(status, upgrading, version) (%s_state)", installationName)),
-					Record: fmt.Sprintf("status:upgrading:version:%s_state:max", installationName),
-				},
-			},
 		},
 	}
-
-	return &resources.AlertReconcilerImpl{
-		ProductName:  "installation",
-		Installation: installation,
-		Log:          log,
-		Alerts:       alerts,
+	err = apiClient.Delete(ctx, metricMissingAlert)
+	if err != nil {
+		return err
```

Comment:
```suggestion
	if err != nil && !k8serr.IsNotFound(err) {
		return err
```
I think we should update the error check because I did see `not found` errors in the logs after the PrometheusRules had been deleted.

Reply:
yep agree